### PR TITLE
Classes for generic types [14315]

### PR DESCRIPTION
**Pull request by jparisu**
After branch #20

### CODE REVIEW

File: amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (juanlofer-eprosima) [inline amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp:1]:
```suggestion
// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

Comment (juanlofer-eprosima) [reply amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp:1]: Not sure if this should be updated for autogenerated files. Anyway, apply the same criteria for all of them.

File: amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    //! \c is_bounded method overriden from \c TopicDataType
+    virtual bool is_bounded() const override;
+
+    //! \c is_plain method overriden from \c TopicDataType
+    virtual bool is_plain() const override;
+
+    //! \c construct_sample method overriden from \c TopicDataType
+    virtual bool construct_sample(
+            void* memory) const override;
```
Comment (juanlofer-eprosima) [inline amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp:105]: These were causing trouble and might not be required, just FYI.

File: amlip_cpp/src/cpp/types/InterfaceDataType.hpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+     * @warning this method must be overriden in child class.
+     */
+    static const char* type_name();
+};
+
+} /* namespace types */
+} /* namespace amlip */
+} /* namespace eprosima */
+
+#endif // AMLIPCPP_TYPES_INTERFACEDATATYPE_HPP
```
Comment (juanlofer-eprosima) [inline amlip_cpp/src/cpp/types/InterfaceDataType.hpp:166]: EOL missing

File: amlip_cpp/src/cpp/types/InterfaceDataType.hpp [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+namespace amlip {
+namespace types {
+
+/**
+ * @brief Interface to create DDS DataTypes.
+ *
+ * Every DataType that will work beneath the AMLIP library must implement this interface.
+ * This is because the "PubSub" type (actually TopicDataType) is a template class, and it needs
+ * to know the type of the data that will be published.
+ * This specialization could only be done from a class inherinting from this interface.
```
Comment (juanlofer-eprosima) [inline amlip_cpp/src/cpp/types/InterfaceDataType.hpp:39]:
```suggestion
 * This specialization could only be done from a class inheriting from this interface.
```

File: amlip_cpp/src/cpp/types/InterfaceDataType.hpp [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+ *
+ * @warning This interface requires to override some methods, including static ones.
+ * Every class inheriting from it must override every method intialized in this file.
```
Comment (juanlofer-eprosima) [inline amlip_cpp/src/cpp/types/InterfaceDataType.hpp:42]:
```suggestion
 * Every class inheriting from it must override every method initialized in this file.
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Uncrustify

File: amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp (hunk 2 above)
Comment (jparisu) [reply amlip_cpp/src/cpp/types/AmlipGenericTopicDataType.hpp:105]: They are required indeed. If the problem arises again, we should take an eye on it.

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**